src/lib/netlist/plib/ptypes.h

```diff
@@ -132,242 +133,364 @@ namespace plib
-	template<> struct numeric_limits<FLOAT128>
+	template <>
+	struct is_floating_point<FLOAT128>
 	{
-		static constexpr FLOAT128 max() noexcept
-		{
-			return FLT128_MAX;
-		}
-		static constexpr FLOAT128 lowest() noexcept
-		{
-			return -FLT128_MAX;
-		}
+		static constexpr bool value = true;
+	};
+	template <>
+	struct numeric_limits<FLOAT128>
+	{
+		static constexpr FLOAT128 max() noexcept { return FLT128_MAX; }
+		static constexpr FLOAT128 lowest() noexcept { return -FLT128_MAX; }
 	};
 #endif
 
-	template<unsigned bits>
+	template <unsigned bits>
 	struct size_for_bits
 	{
-		static_assert(bits <= 64 || (bits <= 128 && compile_info::has_int128::value), "not supported");
-		enum { value =
-			bits <= 8       ?   1 :
-			bits <= 16      ?   2 :
-			bits <= 32      ?   4 :
-			bits <= 64      ?   8 :
-							   16
+		static_assert(bits <= 64
+						  || (bits <= 128 && compile_info::has_int128::value),
+					  "not supported");
+		enum
+		{
+			value = bits <= 8    ? 1
+					: bits <= 16 ? 2
+					: bits <= 32 ? 4
+					: bits <= 64 ? 8
+								 : 16
 		};
 	};
 
-	template<unsigned N> struct least_type_for_size;
-	template<> struct least_type_for_size<1> { using type = uint_least8_t; };
-	template<> struct least_type_for_size<2> { using type = uint_least16_t; };
-	template<> struct least_type_for_size<4> { using type = uint_least32_t; };
-	template<> struct least_type_for_size<8> { using type = uint_least64_t; };
-	template<> struct least_type_for_size<16> { using type = UINT128; };
+	template <unsigned N>
+	struct least_type_for_size;
+	template <>
+	struct least_type_for_size<1>
+	{
+		using type = uint_least8_t;
+	};
+	template <>
+	struct least_type_for_size<2>
+	{
+		using type = uint_least16_t;
+	};
+	template <>
+	struct least_type_for_size<4>
+	{
+		using type = uint_least32_t;
+	};
+	template <>
+	struct least_type_for_size<8>
+	{
+		using type = uint_least64_t;
+	};
+	template <>
+	struct least_type_for_size<16>
+	{
+		using type = UINT128;
+	};
 
 	// This is different to the standard library. Mappings provided in stdint
 	// are not always fastest.
-	template<unsigned N> struct fast_type_for_size;
-	template<> struct fast_type_for_size<1> { using type = uint32_t; };
-	template<> struct fast_type_for_size<2> { using type = uint32_t; };
-	template<> struct fast_type_for_size<4> { using type = uint32_t; };
-	template<> struct fast_type_for_size<8> { using type = uint_fast64_t; };
-	template<> struct fast_type_for_size<16> { using type = UINT128; };
-
-	template<unsigned bits>
+	template <unsigned N>
+	struct fast_type_for_size;
+	template <>
+	struct fast_type_for_size<1>
+	{
+		using type = uint32_t;
+	};
+	template <>
+	struct fast_type_for_size<2>
+	{
+		using type = uint32_t;
+	};
+	template <>
+	struct fast_type_for_size<4>
+	{
+		using type = uint32_t;
+	};
+	template <>
+	struct fast_type_for_size<8>
+	{
+		using type = uint_fast64_t;
+	};
+	template <>
+	struct fast_type_for_size<16>
+	{
+		using type = UINT128;
+	};
```

Comment:
This is less readable in this form – you can’t easily see everything at a glance.

Reply:
Agree that is a case for `clang-format off`